app/cdap/components/Cloud/Profiles/DetailView/Content/ProfileAssociations/index.js

```diff
@@ -223,22 +224,30 @@ export default class ProfileAssociations extends Component {
   };
 
   renderGridBody = () => {
-    let { associationsMap } = this.state;
+    const { associationsMap } = this.state;
     return (
       <div className="grid-body">
         {Object.keys(associationsMap).map((app) => {
-          let appObj = associationsMap[app];
-          let onedayMetrics = objectQuery(appObj, 'metadata', ONEDAYMETRICKEY) || {};
-          let overallMetrics = objectQuery(appObj, 'metadata', OVERALLMETRICKEY) || {};
-          let pipelineUrl = window.getHydratorUrl({
+          const appObj = associationsMap[app];
+          // type === Worker and program === DeltaWorker corresponds to an instance of replicator
+          const isReplicator =
+            appObj.metadata.type === 'Worker' && appObj.metadata.program === 'DeltaWorker';
```

Comment:
Nit: Make this a helper function in `Replicator/utilities`?

Reply:
hmm if thats the case maybe it should be in metadata or something since this data is returned from metadata/search?